sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -649,7 +649,7 @@ struct _pi_context : _pi_object {
   // Mutex for the immediate command list. Per the Level Zero spec memory copy
   // operations submitted to an immediate command list are not allowed to be
   // called from simultaneous threads.
-  std::mutex ImmediateCommandListMutex;
+  pi_shared_mutex ImmediateCommandListMutex;
```

Comment:
You changed it from "mutex" to "shared_mutex". Which one does this really need to be? What I asked before is to wrap "std::mutex" into "pi_mutex", similar to how "std:shared_mutex" is wrapped with "pi_shared_mutex"

Reply:
We can lock pi_shared_mutex for exclusive and  shared access and it is all we need, sorry, but I don't understand the value of implementing wrapper around std::mutex.

It needs to be pi_shared_mutex because we want ability to disable locks for single-threaded mode.

Reply:
> I don't understand the value of implementing wrapper around std::mutex.

The same as in pi_shared_mutex, disable locks in ST mode.
We can use pi_shared_mutex if its performance is the same as std::mutex
Note, there are other places where we still use std::mutex, so let's change them as well

Reply:
Ok, I will add a wrapper around std::mutex as well

Reply:
Fixed.